services/rotor/src/lib/redis.ts

```diff
@@ -18,29 +18,63 @@ function hideSensitiveInfoFromURL(url: string) {
   return parsed.toString();
 }
 
-export function createRedis(): Redis {
-  const redisUrl = requireDefined(process.env.REDIS_URL, "env REDIS_URL is not defined");
-  log.atDebug().log(`Building redis client for ${hideSensitiveInfoFromURL(redisUrl)}`);
-  let tls: any = undefined;
+function resolveRedisConnectionOptions(
+  redisUrl: string,
+  redisSentinelAddress: string | undefined
+): Record<string, any> {
+  let sentinels: any;
+
+  if (redisSentinelAddress) {
+    sentinels = redisSentinelAddress.split(",").map(sentinel => {
+      const [host, port] = sentinel.split(":");
+      return {
+        host,
+        port: port ? parseInt(port, 10) : undefined,
+      };
+    });
+  }
+
+  let tls: any;
   if (redisUrl.startsWith("rediss://")) {
     tls = {
       rejectUnauthorized: false,
     };
   }
-  const redisClient = new Redis(requireDefined(process.env.REDIS_URL, "env REDIS_URL is not defined"), {
-    maxRetriesPerRequest: 3,
-    tls: tls,
-    lazyConnect: false,
-    enableAutoPipelining: true,
-  });
+
+  return {
+    tls,
+    sentinels,
+  };
+}
+
+/**
+ * Example `REDIS_URL` values for a standalone Redis instance:
+ * - Standalone Redis: `redis://username:password@localhost:6379/1`
+ * - Standalone Redis with SSL: `rediss://username:password@localhost:6379/2`
+ *
+ * Example `REDIS_URL` and `REDIS_SENTINEL_ADDRESS` values for Redis Sentinel:
+ * - Redis URL: `redis://username:password@/3?name=mymaster`
+ * - Redis Sentinel Address: `sentinel1:26379,sentinel2:26379,sentinel3:26379`
```

Comment:
A separate `REDIS_SENTINEL_ADDRESS` env variable is required because `ioredis` doesn't support configuring Redis Sentinel directly through the connection URL. Redis Sentinel requires a list of sentinel addresses to be passed separately.